src/rules/exact-npm-version.ts

```diff
@@ -0,0 +1,131 @@
+import { FileNotFoundError } from './../errors/FileNotFoundError';
+import { RuleRegister } from './rule-register';
+import * as fs from 'fs';
+import { FileNotReadableError } from '../errors/FileNotReadableError';
+
+/**
+ * This implementation of Rule modifies Semver in npm's package.json and removes tilds and circumflex
+ * accent in Semver of every dependency.
+ */
+@RuleRegister.register
+export default class ExactNpmVersion {
+  readonly requiredFiles: string[] = ['package.json'];
+  readonly rootPath: string;
+  private packageJSONPath: string;
+  private parsedFile: any;
+  // tslint:disable-next-line: max-line-length
+  readonly semverRegex = /^(\^|\~)((([0-9]+)\.([0-9]+)\.([0-9]+)(?:-([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?)(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?)$/g;
+  readonly jsonObjectsToCheck: string[] = ['dependencies', 'devDependencies'];
```

Comment:
NPM support other types of dependencies : bundledDependencies, optionalDependencies, peerDependencies